File: leetcode/#191.py
# question --> https://leetcode.com/problems/number-of-1-bits/

class Solution:
    def hammingWeight(self, n: int) -> int:
        setBit = 0
        while n: 
            if n & 1: 
                setBit += 1
            n >>= 1
        return setBit
                
        # TC: O(logN); SC: O(1) 

File: leetcode/#191.js
// question --> https://leetcode.com/problems/number-of-1-bits/

/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
    let setBit = 0; 
    while (n) {
        n = n & (n - 1);
        setBit += 1;
    }
    return setBit; 
};

// TC: O(logN); SC: O(1)

// end